### PR TITLE
Allow read records from all layers

### PR DESCRIPTION
**Pull request by alanblins**
Return array of record from all layers. Previous code only returned records from first layer.

### CODE REVIEW

**Comment by gasparesganga** [comment]
A little clarification before discussing any further: there are no _layers_ in a shapefiles, but _geometries_ (or, as used in the official spec, _records_). You are actually talking about records here, not layers.

This change goes completely against the original idea behing the library: reading an arbitrary-sized shapefile without needing a lot of memory, in a fashion similar to `PDOStatement::fetch`.

A typical use case for this library is _fetch the records from the shapefiles and insert them in a database_, where a record-by-record behaviour is desirable.

Furthermore, reading all the records in an Array using the current `getRecord` method is trivial.

In conclusion, _previous code_ was just fine.
